### PR TITLE
Adding libvdwxc (v.0.4.0)

### PR DESCRIPTION
**Pull request by gdonval**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source. **The launchpad repo is the official source package distribution channel.**
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by gdonval** [comment]
Hi @conda-forge/help-c-cpp I believe this is ready for review.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -12,4 +12,3 @@ pr:
 jobs:
   - template: ./.azure-pipelines/azure-pipelines-linux.yml
   - template: ./.azure-pipelines/azure-pipelines-osx.yml
-  - template: ./.azure-pipelines/azure-pipelines-win.yml
```
Comment (wolfv) [inline azure-pipelines.yml:15]: this looks wrong.

File: recipes/libvdwxc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  host:
+    - {{ mpi }}  # [mpi != 'nompi']
+    - fftw * {{ mpi_prefix }}_*
+
+
+test:
+  requires:
+    - python >= 3
+  commands:
+    - test -f ${PREFIX}/lib/libvdwxc.a || exit 1  # [unix]
```
Comment (wolfv) [inline recipes/libvdwxc/meta.yaml:54]: static libraries are discouraged. Is it possible to build shared libraries?

File: recipes/libvdwxc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    - python -c "import ctypes; ctypes.cdll[r'${PREFIX}/lib/libvdwxc${SHLIB_EXT}']"  # [unix]
+    - python -c "import ctypes; ctypes.cdll[r'${PREFIX}/lib/libvdwxcfort${SHLIB_EXT}']"  # [unix]
+
+    
+about:
+  home: https://libvdwxc.org
+  dev_url: https://gitlab.com/libvdwxc/libvdwxc
+  doc_url: https://libvdwxc.org/index.html
+  license: GPL-3.0-or-later
+  license_url: https://opensource.org/licenses/GPL-3.0
```
Comment (wolfv) [inline recipes/libvdwxc/meta.yaml:68]: license_url and license_file are not necessary when using SPDX ids

File: recipes/libvdwxc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+build:
+  skip: True  # [not unix]
+  number: {{ build_number }}
+  string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build_number }}"
+  run_exports:
+    - {{ pin_compatible('numpy', min_pin='x.x', max_pin='x.x') }}
+    {% if mpi != 'nompi' %}
+    - {{ name }} * {{ mpi_prefix }}_*
+    {% endif %}
+  script:
```
Comment (wolfv) [inline recipes/libvdwxc/meta.yaml:29]: maybe it would be better to write an external `build.sh` script.

File: azure-pipelines.yml (hunk 1 above)
Comment (gdonval) [reply azure-pipelines.yml:15]: (Keep in mind this is my first recipe, so I'm bound to ask dumb questions)
 
What's the way to handle unsupported platforms then? Is it enough if no artefact is generated? Still wasteful of CI time though.

Comment (wolfv) [reply azure-pipelines.yml:15]: yes, we waste some CI but only in the staged-recipes. This is taken care of in the generated "feedstock".

Merging this would destroy the pipeline for everyone coming after you.

File: recipes/libvdwxc/meta.yaml (hunk 2 above)
Comment (gdonval) [reply recipes/libvdwxc/meta.yaml:54]: Shared libraries **are** already generated too so should I split this package into `libvdwxc` and `libvdwxc-static` then?

Comment (wolfv) [reply recipes/libvdwxc/meta.yaml:54]: Or just remove the static library? Do you need it?

File: azure-pipelines.yml (hunk 1 above)
Comment (gdonval) [reply azure-pipelines.yml:15]: Oh I see, I'm sorry: I'll revert that when I get to the office!

File: recipes/libvdwxc/meta.yaml (hunk 2 above)
Comment (gdonval) [reply recipes/libvdwxc/meta.yaml:54]: The topic is rather complex. For the time being, I'll remove them and if users are really adamant about getting static libs, I'll create a static lib variant.

**Comment by gdonval** [comment]
@wolfv I believe all is good now.

**Comment by gdonval** [comment]
@conda-forge-admin, please restart ci

**Comment by carterbox** [comment]
> WARNING (libvdwxc): run-exports library package conda-forge::libgfortran5-9.3.0-h6c81a4c_23 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)

I'm not too familiar with compiling combined fortran/c libraries, but conda-build is reporting that this library (which is added when you ask for the fortran compiler) may not be used. Are you are using it?

**Comment by gdonval** [comment]
The compiler (and libraries) is definitely used for the tests. 

Besides that, the only fortran part is the fortran-C interface and I don't think a compiler is necessary for those (they should be similar to include files). Should I try to compile the tests in testing phase or is there a way to tell conda not to include libgfortran as a dependency?

File: recipes/libvdwxc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+build:
+  skip: True  # [not unix]
+  number: {{ build_number }}
+  string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build_number }}"
+  run_exports:
+    {% if mpi != 'nompi' %}
+    - {{ name }} * {{ mpi_prefix }}_*
+    {% endif %}
+
```
Comment (carterbox) [inline recipes/libvdwxc/meta.yaml:27]:
```suggestion
  ignore_run_exports:
    - libgfortran5
 
```
This is how you ignore run dependencies from packages in build or host.

File: recipes/libvdwxc/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  sha256: 3524feb5bb2be86b4688f71653502146b181e66f3f75b8bdaf23dd1ae4a56b33
+
+build:
+  skip: True  # [not unix]
+  number: {{ build_number }}
+  string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build_number }}"
+  run_exports:
+    {% if mpi != 'nompi' %}
+    - {{ name }} * {{ mpi_prefix }}_*
+    {% endif %}
```
Comment (carterbox) [inline recipes/libvdwxc/meta.yaml:26]: Are run exports not needed when using the `nompi` variant?
```suggestion
  run_exports:
    {% if mpi != 'nompi' %}
    - {{ name }} * {{ mpi_prefix }}_*
    {% endif %}
    - {{ pin_subpackage('libvdwxc', max_pin='x.x') }}
```

Comment (gdonval) [reply recipes/libvdwxc/meta.yaml:26]: From my (still) limited understanding of run_exports, it looks like the condition is not necessary indeed. I copied what had been done in other similar packages. I suppose the main concern of those other packages was not mixing mpi implementations and if the 'nompi' version is the default anyway, all should work. **But**, if removing the condition just make things more reliable, I don't see why not removing it.

Comment (gdonval) [reply recipes/libvdwxc/meta.yaml]: Anyway, I have removed it.

**Comment by gdonval** [comment]
@carterbox I believe all is OK now.

File: recipes/libvdwxc/meta.yaml (hunk 6 above)
Comment (carterbox) [reply recipes/libvdwxc/meta.yaml]: > From my (still) limited understanding of run_exports

run_exports tells conda that "If libvdwxc is in the build/host section, then you need to put libvdwxc in your run requirements too" because any package that links against libvdwc needs to have it at runtime. (As opposed to something like CMAKe which is only a tool and doesn't link to the package).

> if the 'nompi' version is the default anyway

There is no "default" There is only what the environment solver finds to be the first package which meets the requirements.

> if removing the condition just make things more reliable

By removing `{% if mpi != 'nompi' %}` you are saying that libraries compiled against libvdwxc-nompi must be run with libvdwxc-nompi. As opposed to *any* libvdwxc. Is that what you want?

My suggested change only adds `- {{ pin_subpackage('libvdwxc', max_pin='x.x') }}` because I think that compiling against libvdwc of any *variant* requires libvdwc of the same version *number* (not variant). 

tl;dr If it's OK to use libvdwc-mpich when you compiled against libvdwc-nompi, then put the `{% if mpi != 'nompi' %}` back.

Comment (gdonval) [reply recipes/libvdwxc/meta.yaml]: All this makes sense, thanks for the explanation. 

Version number is important, therefore pinning was a good idea.

We should be in the case you described in the last paragraph (for compilation only). At runtime, if the calling program has no MPI awareness or vice-versa, things will definitely go wrong so I'm keen on leaving the condition out.

Comment (carterbox) [reply recipes/libvdwxc/meta.yaml]: Thanks for your patience.
